File: software/examples/vbnet/ExampleButtonCallback.vb
Imports System
Imports Tinkerforge

Module ExampleButtonCallback
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change to your UID

    ' Callback subroutine for button pressed callback
    Sub ButtonPressedCB(ByVal sender As BrickletLCD16x2, ByVal button As Byte)
        Console.WriteLine("Button Pressed: " + button.ToString())
    End Sub

    ' Callback subroutine for button released callback
    Sub ButtonReleasedCB(ByVal sender As BrickletLCD16x2, ByVal button As Byte)
        Console.WriteLine("Button Released: " + button.ToString())
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim lcd As New BrickletLCD16x2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Register button pressed callback to subroutine ButtonPressedCB
        AddHandler lcd.ButtonPressed, AddressOf ButtonPressedCB

        ' Register button released callback to subroutine ButtonReleasedCB
        AddHandler lcd.ButtonReleased, AddressOf ButtonReleasedCB

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/delphi/ExampleButtonCallback.pas
program ExampleButtonCallback;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletLCD16x2;

type
  TExample = class
  private
    ipcon: TIPConnection;
    lcd: TBrickletLCD16x2;
  public
    procedure ButtonPressedCB(sender: TBrickletLCD16x2; const button: byte);
    procedure ButtonReleasedCB(sender: TBrickletLCD16x2; const button: byte);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'XYZ'; { Change XYZ to the UID of your LCD 16x2 Bricklet }

var
  e: TExample;

{ Callback procedure for button pressed callback }
procedure TExample.ButtonPressedCB(sender: TBrickletLCD16x2; const button: byte);
begin
  WriteLn(Format('Button Pressed: %d', [button]));
end;

{ Callback procedure for button released callback }
procedure TExample.ButtonReleasedCB(sender: TBrickletLCD16x2; const button: byte);
begin
  WriteLn(Format('Button Released: %d', [button]));
end;

procedure TExample.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  lcd := TBrickletLCD16x2.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Register button pressed callback to procedure ButtonPressedCB }
  lcd.OnButtonPressed := {$ifdef FPC}@{$endif}ButtonPressedCB;

  { Register button released callback to procedure ButtonReleasedCB }
  lcd.OnButtonReleased := {$ifdef FPC}@{$endif}ButtonReleasedCB;

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.
